src/portal/src/app/project/tag-retention/tag-retention.component.ts

```diff
@@ -62,14 +70,36 @@ export class TagRetentionComponent implements OnInit {
     loadingExecutions: boolean = false;
     loadingHistories: boolean = false;
     dryRun: boolean = false;
+    label: string = 'TAG_RETENTION.TRIGGER';
+    loadingRule: boolean = false;
+    @ViewChild('cronScheduleComponent')
+    cronScheduleComponent: CronScheduleComponent;
     @ViewChild('addRule') addRuleComponent: AddRuleComponent;
-
     constructor(
         private route: ActivatedRoute,
         private tagRetentionService: TagRetentionService,
         private errorHandler: ErrorHandler,
     ) {
     }
+    originCron(): OriginCron {
+        let originCron: OriginCron = {
+            type: "Daily",
+            cron: "0 0 0 * * *"
+        };
+        originCron.cron = this.retention.trigger.settings.cron;
```

Comment:
Almost the same as cron-schedule.component Can you reuse that one? If not , please abstract a utility service share with cron-schedule.component.

Reply:
Yes, it can be reused base on  confirming with Steven Zou